contracts/RestrictedNORI.sol

```diff
@@ -0,0 +1,1089 @@
+      }
+    }
+    return super._beforeTokenTransfer(operator, from, to, ids, amounts, data);
+  }
+
+  /**
+   * Linearly released balance for a single schedule at the current block timestamp, ignoring any
+   * released amount floor that has been set for the schedule.
+   */
+  function _linearReleaseAmountAvailable(uint256 scheduleId)
+    internal
+    view
+    returns (uint256)
+  {
+    Schedule storage schedule = _scheduleIdToScheduleStruct[scheduleId];
+    if (block.timestamp >= schedule.endTime) {
+      return totalSupply(scheduleId);
+    }
+    uint256 rampTotalTime = schedule.endTime - schedule.startTime;
+    return
+      block.timestamp < schedule.startTime
+        ? 0
+        : (totalSupply(scheduleId) * (block.timestamp - schedule.startTime)) /
+          rampTotalTime;
+  }
+
+  /**
+   * @dev Calculates the number of tokens that should be revoked from a given token holder and schedule based on their
+   * proportion of ownership of that schedule's tokens and the total number of tokens being revoked.
+   */
+  function _quantityToRevokePerTokenHolder(
+    uint256 totalQuantityToRevoke,
+    uint256 scheduleId,
+    address account
+  ) private view returns (uint256) {
+    Schedule storage schedule = _scheduleIdToScheduleStruct[scheduleId];
+    uint256 scheduleTrueTotal = schedule.totalClaimedAmount +
+      totalSupply(scheduleId);
+    // avoid division by 0
+    if (scheduleTrueTotal == 0) {
+      return 0;
+    }
+    uint256 balanceOfAccount = balanceOf(account, scheduleId);
+    uint256 claimedAmountForAccount = schedule.claimedAmountsByAddress[account];
+    uint256 revocableForAccount = ((claimedAmountForAccount +
+      balanceOfAccount) * (totalQuantityToRevoke)) / scheduleTrueTotal;
+
+    return revocableForAccount;
+  }
+
+  /**
+   * @dev The total amount of released tokens available at the current block timestamp for the schedule.
+   * Takes the maximum of either the calculated linearly released amount based on the schedule parameters,
+   * or the released amount floor, which is set at the current released amount whenever the balance of a
+   * schedule is decreased through revocation or withdrawal.
+   */
+  function _releasedBalanceOfSingleSchedule(uint256 scheduleId)
+    internal
+    view
+    returns (uint256)
+  {
+    Schedule storage schedule = _scheduleIdToScheduleStruct[scheduleId];
+    return
+      MathUpgradeable.max(
+        _linearReleaseAmountAvailable(scheduleId),
+        schedule.releasedAmountFloor
+      );
+  }
+
+  /**
+   * Grants `role` to `account`
+   *
+   * ### Requirements:
+   *    - SCHEDULE_CREATOR_ROLE cannot be granted to an owner of RestrictedNORI tokens
+   *
+   * [See OZ Access Control docs for more] (
+   *  https://docs.openzeppelin.com/contracts/3.x/access-control)
+   */
+  function _grantRole(bytes32 role, address account) internal virtual override {
+    if (role == SCHEDULE_CREATOR_ROLE) {
+      if (_addressToScheduleIdSet[account].length() > 0) {
```

Comment:
why is this needed?

Reply:
Unless I'm missing something I think this override should be cut

Reply:
I think you're right?  This is basically preventing anyone who already has ownership of a schedule from being able to become a schedule creator. (a  pattern in `LockedNORI`)  but I don't think it matters here... I just don't really see what security concern it's really preventing

Reply:
removed